### ISSUE TITLE
Value Conversions not working with IList<T> property

### ISSUE THREAD

**Issue by Mergasov** [aspnet/EntityFrameworkCore#11501]
My model class have a  `IList<string>` property. I use value conversions to save it as json-string in DB. SaveChanges operation throws the `System.InvalidCastException`.
When I change the type of the property to `List<string>` no exception occurs. Is this behavior by design or a bug? It's important for me to have a property of the interface type.

**Update**: as I understand there is a comparsion of `typeof(IList<string>)` and `ListValue.GetType()` [under the hood](https://github.com/aspnet/EntityFrameworkCore/blob/a7e4a9e7ec8579510a423168d205943dc7422083/src/EFCore/Storage/ValueConversion/ValueConverter%60.cs#L52) so this returns false when any interface type was declared on entity class (and calling `Convert.ChangeType` as a consequence). Any planning to ease this restriction?

**Exception message:**
`Microsoft.EntityFrameworkCore.DbUpdateException: An error occurred while updating the entries. See the inner exception for details. ---> System.InvalidCastException: Object must implement IConvertible`

**Stack trace:**
```
   at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
   at Microsoft.EntityFrameworkCore.Storage.Converters.ValueConverter`2.Sanitize[T](Object value)
   at Microsoft.EntityFrameworkCore.Storage.Converters.ValueConverter`2.<>c__DisplayClass1_0`2.<SanitizeConverter>b__0(Object v)
   at Microsoft.EntityFrameworkCore.Storage.RelationalTypeMapping.CreateParameter(DbCommand command, String name, Object value, Nullable`1 nullable)
   at Microsoft.EntityFrameworkCore.Storage.Internal.TypeMappedRelationalParameter.AddDbParameter(DbCommand command, Object value)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalParameterBase.AddDbParameter(DbCommand command, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.CreateCommand(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.<ExecuteAsync>d__17.MoveNext()
```

### Steps to reproduce
```c#
public abstract class Animal
{
    [NotMapped]
    public abstract IList<string> Names { get; set; }
}

public class Cat : Animal
{
      public override IList<string> Names { get; set; }
}

//DbContext class
using Newtonsoft.Json;

protected override void OnModelCreating(ModelBuilder builder)
{
    builder.Entity<Animal>()
                .HasDiscriminator<string>("type")
                .HasValue<Cat>("cat");

   builder.Entity<Cat>()
                .Property(t => t.Names)
                .HasConversion(
                  v => JsonConvert.SerializeObject(v),
                  v => JsonConvert.DeserializeObject<IList<string>>(v))
                .IsRequired();
}

public DbSet<Animal> Animals { get; set; }

//main
Animal animal = MyAbstractFactory.GetAnimal();
animal.Names = new List<string> { "murka" };
dbContext.Animals.Add(animal);
dbContext.SaveChanges(); // throws here
```

### Further technical details
EF Core version: 2.1.0-preview1-final
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 7
IDE: Visual Studio 2017 15.5.2

**Comment by divega** [comment]
@Mergasov could you try our nightly builds? We believe we have fixed this for preview 2.

**Comment by Mergasov** [comment]
@divega  got the same exception using preview2-28177

**Comment by divega** [comment]
Thank you @Mergasov. 

@AndriySvyryd, would you be the right person to investigate this now?

**Comment by divega** [comment]
@Mergasov thanks for raising this issue. We have fixed it but the fix will be available in a build after 2.1-preview2. In the meanwhile you can use the class `List<T>`.
